Add styles for CRUD example

scabbiaza opened the pull request:
Update webpack config in CRUD example: keep configs only for FE
Note: `new ExtractTextPlugin` in plugins makes merging problematic

scabbiaza committed:
Add styles for CRUD example

Update webpack config in CRUD example: keep configs only for FE
Note: `new ExtractTextPlugin` in plugins makes merging problematic

scabbiaza commented:
Decided to separate filters and sorts by the linebreak, otherwise `users` list didn't look good.

<img width="760" alt="screen shot 2017-11-29 at 19 24 43" src="https://user-images.githubusercontent.com/2734464/33389287-0648f26a-d53b-11e7-9e06-625004952c26.png">

<img width="1081" alt="screen shot 2017-11-29 at 19 24 50" src="https://user-images.githubusercontent.com/2734464/33389289-0872c99e-d53b-11e7-9df0-983992752700.png">

scabbiaza committed:
Update gitignore

Exclude the whole `public` folder from CRUD example

scabbiaza committed:
Update CRUD example

1. Refactor `public` folder: remove `css` and `js` folder from public.
   Motivation: `public` folder should have similar structure to the app folder. With loading on demand option scripts and assets will be per folder.
2. Add an example of using styles and images in nested components
3. Styles and images should be in the separate folders and not at the same folder where `js` files
4. Exclude Assets files from importing in SSR

ivan-kleshnin commented on .gitignore:
```diff
@@ -4,4 +4,4 @@
 npm-debug.log*
 */*/public/js
-*/*/public/css
+examples/7.crud/public
```
Why?

ivan-kleshnin commented on examples/7.crud/client/home/PostIndex.js:
```diff
@@ -3,9 +3,13 @@ import * as R from "ramda"
 import React from "react"
 import PostItem from "./PostItem"
 
+// in order to Webpack move this file to `public` folder
```
Also to minimize and possibly inject them as base-64 (for images).

ivan-kleshnin commented on examples/7.crud/webpack.config.js:
```diff
@@ -43,7 +51,7 @@ let configs = {
   },
   plugins: [
-    new ExtractTextPlugin("css/[name].css"),
+    new ExtractTextPlugin("[name].css"),
```
Why `/css/` and `/js/` suffixes are removed?

scabbiaza committed:
Update CRUD example

1. Rename `app.css` -> `bundle.css`, `app.js` -> `bundle.js`
2. Remove `styles` and `images` folder

ivan-kleshnin commented:
Great, thanks!
